pkg/detectors/zulipchat/zulipchat.go

```diff
@@ -12,24 +12,26 @@ import (
 	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
 )
 
-type Scanner struct{}
+type Scanner struct{
+	client *http.Client
+}
 
 // Ensure the Scanner satisfies the interface at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	client = common.SaneHttpClient()
+	defaultClient = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat    = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat"}) + common.BuildRegex(common.AlphaNumPattern, "", 32))
-	idPat     = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat"}) + common.EmailPattern)
-	domainPat = regexp.MustCompile(detectors.PrefixRegex([]string{"zulipchat", "domain"}) + common.SubDomainPattern)
+	keyPat    = regexp.MustCompile(common.BuildRegex(common.AlphaNumPattern, "", 32))
+	idPat     = regexp.MustCompile(`\b(([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-]+)(\.[a-zA-Z]{2,5}){1,2})\b`)
+	domainPat = regexp.MustCompile(`\b([a-z0-9-]+\.zulipchat\.com)\b`)
```

Comment:
IMO, this should include both `zulip.com` and `zulip.org` to handle all potential cases.

```suggestion
       // I think this is right. Ought to be double-checked though.
	domainPat = regexp.MustCompile(`\b([a-z0-9-]+\.zulip(?:(?:chat)?\.com|\.org))\b`)
```

https://github.com/alfonsrv/impf-botpy/blob/2eb10199984b3a575f645891003ba4b7afe3ecae/docs/ALERTS.md?plain=1#L52
https://github.com/zulip/zulip-js/blob/db9b0e4f7535552e768d6ecc7b3f4fbf4ea910fe/examples/zuliprc#L3